athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchMetadataHandler.java

```diff
@@ -292,30 +308,35 @@ public GetSplitsResponse doGetSplits(BlockAllocator allocator, GetSplitsRequest
         Set<Split> splits = new HashSet<>();
         // Get domain
         String domain = request.getTableName().getSchemaName();
-        // Get index
-        String index = request.getTableName().getTableName();
 
         try {
             String endpoint = getDomainEndpoint(domain);
             AwsRestHighLevelClient client = clientFactory.getOrCreateClient(endpoint);
+            // We send index request in case the table name is a data stream, a data stream can contains multiple indices which are created by ES
+            // For non data stream, index name is same as table name
+            GetIndexResponse indexResponse = client.indices().get(new GetIndexRequest(request.getTableName().getTableName()), RequestOptions.DEFAULT);
+            // Get index list, iterate thru all index, if
+            String[] indices = indexResponse.getIndices();
             try {
-                Set<Integer> shardIds = client.getShardIds(index, queryTimeout);
-                for (Integer shardId : shardIds) {
-                    // Every split must have a unique location if we wish to spill to avoid failures
-                    SpillLocation spillLocation = makeSpillLocation(request);
-                    // Create a new split (added to the splits set) that includes the domain and endpoint, and
-                    // shard information (to be used later by the Record Handler).
-                    splits.add(new Split(spillLocation, makeEncryptionKey(), ImmutableMap
-                            .of(domain, endpoint, SHARD_KEY, SHARD_VALUE + shardId.toString())));
+                for (String index : indices) {
+                    Set<Integer> shardIds = client.getShardIds(index, queryTimeout);
+                    for (Integer shardId : shardIds) {
+                        // Every split must have a unique location if we wish to spill to avoid failures
+                        SpillLocation spillLocation = makeSpillLocation(request);
+                        // Create a new split (added to the splits set) that includes the domain and endpoint.
+                        // In addition, we retain shard and index information (to be used later by the Record Handler)
+                        splits.add(new Split(spillLocation, makeEncryptionKey(), ImmutableMap
+                                .of(domain, endpoint, SHARD_KEY, SHARD_VALUE + shardId.toString(), INDEX_KEY, index)));
```

Comment:
Rewrite this function using Streams. It will be simpler and avoids the ITM anti-pattern.